tests/regression_tests/unstructured_mesh/source_sampling/test.py

```diff
@@ -0,0 +1,114 @@
+from itertools import product

```

Comment:
Yeah, let me think about something meaningful to tally here -- probably something with a `CellBorn` filter.

Reply:
I ended up adding a flux tally here. It isn't that meaningful for this model though. It's main purpose is to ensure the `openmc` executable will run successfully with a mesh source -- the unit tests only sample source sites.

Maybe it would be better to do both the source site sampling and a full run as part of the unit tests instead?

Reply:
I went a head and added a call to run particles in the `test_mesh_source.py` to check that transport is successful for a handful of particles. The check for an incorrect strengths array size was moved there as well and I've removed this regression test. I think that makes the most sense for this capability.